apps/src/netsim/NetSimRouterNode.js

```diff
@@ -849,43 +859,25 @@ NetSimRouterNode.prototype.setMemory = function (newMemory) {
 };
 
 /**
- * Query the wires table and pass the callback a list of wire table rows,
- * where all of the rows are wires attached to this router.
- * @param {NodeStyleCallback} onComplete which accepts an Array of NetSimWire.
+ * @returns {NetSimWire[]} all of the wires that are attached to this router.
  */
-NetSimRouterNode.prototype.getConnections = function (onComplete) {
-  onComplete = onComplete || function () {};
-
+NetSimRouterNode.prototype.getConnections = function () {
   var shard = this.shard_;
-  var wireTable = shard.wireTable;
   var routerID = this.entityID;
-  wireTable.refresh()
-    .fail(function (err) {
-        onComplete(err, []);
+  return shard.wireTable.readAll()
+      .filter(function (wireRow) {
```

Comment:
likewise, `return shard.wireTable.readAll().filter(function (wireRow) {` all on one line here is much harder to misread